dashboard/src/actions/tocActions.js

```diff
@@ -0,0 +1,124 @@
+import * as TYPES from "./types";
+
+import API from "utils/axiosInstance";
+import { DANGER } from "assets/constants/toastConstants";
+import { showToast } from "./toastActions";
+import { uriTemplate } from "utils/helper";
+
+/**
+ * Function to fetch contents data
+ * @function
+ * @param {String} param - Dataset ID
+ * @param {String} dataUri - URI
+ * @param {String} item - Active item
+ * @param {Boolean} isSubDir - To identify sub-directory expansion
+ * @return {Function} - dispatch the action and update the state
+ */
+export const fetchTOC =
+  (param, dataUri, item, isSubDir) => async (dispatch, getState) => {
+    try {
+      dispatch({ type: TYPES.LOADING });
+      const endpoints = getState().apiEndpoint.endpoints;
+      const parent = dataUri?.split("contents/").pop();
+      const uri = uriTemplate(endpoints, "datasets_contents", {
+        dataset: param,
+        target: parent,
+      });
```

Comment:
I'm not really sure how to feel about this. It seems wasteful to strip the directory path off the URL just to build the identical URL from the resource ID and path, although this does avoid a special case for the initial query where you don't have a URL and just supply an empty string which will "split" into `[""]` and produce the empty path you want.

However, another alternative would be to just fetch the URL, and for the TOC page to make that first call with `uriTemplate(endpoints, "datasets_contents"), {dataset: resource_id, target: ""});`

Reply:
Updated